cdist/exec/remote.py

```diff
@@ -140,8 +140,7 @@ def extract_archive(self, path, mode):
         # AIX only allows -C at the end
         command = [
             "tar",
-            "-x",
-            "-f", path,
+            "xf", path,
             "-C", os.path.dirname(path)
```

Comment:
To be extra careful, given that some OS have been making problems with `-C` already: use `cd path && tar xf" instead?

Reply:
It's not possible to use `[..., "&&", ...]` without major changes. Or I'm not seeing the way.

Reply:
`run()` only `shquot.join()`s the given command if it is a list. From a quick glance at this module I don't see anything stopping us from assembling the shell command as a string in this function and then pass the resulting string to `self.run()`.

Reply:
You're right, not sure what I did before when I tried exactly that and failed. Anyway...